src/main/java/com/xwiki/authentication/saml/samlauth/SamlXwikiAttributesExtractor.java

```diff
@@ -0,0 +1,111 @@
+
+public class SamlXwikiAttributesExtractor {
+    private static final Logger LOG = LoggerFactory.getLogger(SamlAuthenticationHandler.class);
+
+    private final SamlAuthConfig authConfig;
+
+    public SamlXwikiAttributesExtractor(SamlAuthConfig authConfig) {
+        this.authConfig = authConfig;
+    }
+
+    public Saml2XwikiAttributes extractXWikiAttributesFromSaml(Auth auth) {
+        return new SamlXwikiAttributesExtractorWithAuth(auth).extractXWikiAttributesFromSaml();
+    }
+
+    class SamlXwikiAttributesExtractorWithAuth {
+        private final Auth auth;
+
+        public SamlXwikiAttributesExtractorWithAuth(Auth auth) {
+            this.auth = auth;
+        }
+
+        Saml2XwikiAttributes extractXWikiAttributesFromSaml() {
+            final Map<String, String> samlAttributes = extractSamlAttributes();
+
+            final Set<String> groupsFromSaml = newHashSet((defaultIfNull(samlAttributes.get("XWikiGroups"), "")).split(","));
+            groupsFromSaml.add(authConfig.defaultGroupForNewUsers);
+
+            final Map<String, String> xwikiAttributes = mapToXwikiAttributes(samlAttributes);
+            final String nameID = auth.getNameId();
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("SAML ID is [{}]", nameID);
+                LOG.debug("SAML samlAttributes are [{}]", samlAttributes);
+                LOG.debug("SAML user data are [{}]", xwikiAttributes);
+            }
+            return new Saml2XwikiAttributes(nameID, xwikiAttributes, groupsFromSaml);
+        }
+
+        private Map<String, String> extractSamlAttributes() {
+            final Map<String, String> samlAttributes = new HashMap<>();
+
+            try {
+                LOG.debug("Reading authentication response");
+                auth.getAttributesName().forEach(attributeName ->
+                        samlAttributes.put(attributeName, String.join(",", auth.getAttributes().get(attributeName))));
+
+            } catch (Exception e1) {
+                LOG.error("Failed reading authentication response", e1);
+                throw e1;
+            }
+            return samlAttributes;
+        }
+
+        private Map<String, String> mapToXwikiAttributes(Map<String, String> samlAttributes) {
+            final Map<String, String> extInfos = new HashMap<>();
+            for (Map.Entry<String, String> mapping : getSamlToXwikiMapping().entrySet()) {
+                final String samlAttributeValue = samlAttributes.get(mapping.getKey());
+
+                if (samlAttributeValue != null)
+                    extInfos.put(mapping.getValue(), samlAttributeValue);
+            }
+            return extInfos;
+        }
+
+        private Map<String, String> getSamlToXwikiMapping() {
+            final Map<String, String> userPropertiesMapping = new HashMap<>();
+            for (String fieldMapping : authConfig.fieldMapping) {
+                final String[] fieldAndValue = fieldMapping.split("=");
```

Comment:
Add a limit to the split:
```suggestion
                final String[] fieldAndValue = fieldMapping.split("=", 2);
```

This way, you don't error out on values with `=` (Not that I'd imagine a SAML mapping with a `=`, but just in case).